doc/proposals/images-subspec.md

```diff
@@ -0,0 +1,257 @@
+  - '@zachncst'
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-11-04
+last-updated: 2019-11-05
+status: provisional
+help-from:
+  - '@camilamacedo86'
+  - '@kevinrizza'
+  - '@shawn-hurley'
+see-also:
+  - 'https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/contributors/design-proposals/related-images.md'
+---
+
+# Images Sub Spec
+
+## Release Signoff Checklist
+
+- \[ \] Enhancement is `implementable`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created in
+  [operator-sdk/doc][operator-sdk-doc]
+
+## Tag Line
+
+> Who likes hard coding image locations anyways?
+
+## Summary
+
+When an operator is installed, the images that the operator installs as operands in the
+form of pods, deployments or other kubernetes resources is not always clearly
+defined and editable by a user of an operator. This proposal would define
+an operator-sdk ImageSpec and ImagesListSpec object structs that operator developers
+could use to enhance the creation of operator CRDs or improve existing
+operators. The spec objects would let operator users override image locations
+for that custom resource.
+
+Additionally, the proposal would provide tooling to generate scaffold code for
+creating new operators. When creating a new operator, an operator developer
+would be able to pass flags that would create a base spec object that includes
+the ImagesListSpec.
+
+## Motivation
+
+This proposal is driven by a few motivations:
+
+- Makes operators more transparent with what is being installed on the cluster.
+- Formalizes a best practice of operator development.
+- The depencency between operator version and operator resource version is broken.
+- Allows for out of band updates of the underlying images.
+- Registry information for the images can easily be overridden to point at
+  private registries.
+- Provides some (potential) standardization to the CRD.
+
+## Goals
+
+- Provide an easy mechanism of defining and overriding images and related image
+  information (like pull secrets).
+- Increase transparency with what operators are installing.
+- Simple, reusable specs to build operators; similar to PodSpec in Kubernetes.
+
+### Non-Goals
+
+- OLM management of the images spec is not in this proposal.
+- Mapping/re-mapping other container values (ports, etc.).
+- Additional pod information, like env vars.
+
+## Proposal
+
+Implementing the proposal would include:
+
+- Add an optional section to the CRD spec named images that contains definitions of
+  all images used by the operator. An example:
+
+  ```yaml
+  apiVersion: app.example.com/v1alpha1
+  kind: AppService
+  metadata:
+    name: example-appservice
+  spec:
+    size: 3
+    images:
+      - name: nginx
+        image: nginx:1.7.9
+      - name: example
+        iamge: my-registry:5000/example:1.0.0
+  ```
+
```

Comment:
Other reason for this file not be created yet is because of kubebuilder add all API, version kinds generated by in the PROJECT file already as the following example. 

```
version: "2"
domain: my.domain
repo: test
resources:
- group: webapp
  version: v1
  kind: Guestbook
- group: webapp
  version: v1
  kind: Guestbook2
```

So, after the integration, we may decide to use it to add the images as well. For example. 

```
version: "2"
domain: my.domain
repo: test
resources:
- group: webapp
  version: v1
  kind: Guestbook
- group: webapp
  version: v1
  kind: Guestbook2
  image: image:tag 
```